File: HelloPython/d210210/myupdate.py
import cx_Oracle

conn = cx_Oracle.connect('python/python@localhost:1521/xe')

cur = conn.cursor()

sql = "UPDATE SAMPLE SET col02 = :V_col02, col03 = :V_col03 WHERE col01 = :V_col01"
V_col01 = '9'
V_col02 = '8'
V_col03 = '7'
params = (V_col02, V_col03, V_col01)
cur.execute(sql, params)
print(cur.rowcount)

conn.commit()

cur.close()
conn.close()

File: java/java-workspace2/src/day210210/MyUpdate.java
package day210210;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

public class MyUpdate {
	public static void main(String[] args) throws ClassNotFoundException, SQLException {
		Connection con;
		String url = "jdbc:oracle:thin:@localhost:1521:xe";
		Class.forName("oracle.jdbc.driver.OracleDriver");

		con = DriverManager.getConnection(url, "python", "python");

		String sql = "UPDATE SAMPLE SET COL02 = '5', COL03 = '5' WHERE COL01 = '1'";
		Statement stmt = con.createStatement();

		int result = stmt.executeUpdate(sql);
		if (result > 0) {
			System.out.println("성공");
		} else {
			System.out.println("실패");
		}
		stmt.close();
		con.close();
	}
}
